src/random.cpp

```diff
@@ -43,15 +43,24 @@ static void RandFailure()
 
 static inline int64_t GetPerformanceCounter()
 {
-    int64_t nCounter = 0;
-#ifdef WIN32
-    QueryPerformanceCounter((LARGE_INTEGER*)&nCounter);
+    // Read the hardware time stamp counter when available.
+    // See https://en.wikipedia.org/wiki/Time_Stamp_Counter for more information.
+#if defined(_MSC_VER)
+    return __rdtsc();
```

Comment:
MSVC has different macros. I've solved it in a way that seems right, but I can't test on MSVC.